File: src/main/java/com/dogGetDrunk/meetjyou/version/AppVersionDto.java
package com.dogGetDrunk.meetjyou.version;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class AppVersionDto {

    private String version;
    private boolean forceUpdate;
    private String downloadUrl;
    private LocalDateTime releasedAt;

    public static AppVersionDto fromEntity(AppVersion appVersion) {
        return AppVersionDto.builder()
                .version(appVersion.getVersion())
                .forceUpdate(appVersion.isForceUpdate())
                .downloadUrl(appVersion.getDownloadUrl())
                .releasedAt(appVersion.getReleasedAt())
                .build();
    }
}

File: src/main/java/com/dogGetDrunk/meetjyou/version/AppVersionDto.kt
package com.dogGetDrunk.meetjyou.version

import java.time.Instant

data class AppVersionDto(
    val platform: Platform,
    val version: String,
    val forceUpdate: Boolean,
    val message: String? = null,
    val storeReleased: Boolean = false,
    val releasedAt: Instant,
) {
    companion object {
        fun fromEntity(appVersion: AppVersion) = AppVersionDto(
            platform = appVersion.platform,
            version = appVersion.version,
            forceUpdate = appVersion.forceUpdate,
            message = appVersion.message,
            storeReleased = appVersion.storeReleased,
            releasedAt = appVersion.releasedAt,
        )
    }
}
